src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -80,16 +82,28 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
     apiOperations
   )
 
-  /** Inits registered tokens at startup */
+  /** Inits tables with values from CSV files */
   import kantan.csv.generic._
-
-  val tokenRows: List[Tables.RegisteredTokensRow] =
-    ConfigUtil.Csv.readTableRowsFromCsv(Tables.RegisteredTokens, tezosConf.network, separator = '|')
-
-  db.run(TezosDb.writeRegisteredTokensRowsIfEmpty(tokenRows)) andThen {
-      case Success(_) => logger.info(s"Written ${tokenRows.size} registered token rows")
-      case Failure(e) => logger.error("Could not fill registered_tokens table", e)
+  initTableFromCsv(Tables.RegisteredTokens, tezosConf.network, separator = '|')
+  initTableFromCsv(Tables.KnownAddresses, tezosConf.network)
+
+  import shapeless._
+  import shapeless.ops.hlist._
+
+  import kantan.csv._
+
+  /** Reads and inserts CSV file to the database for the given table */
+  def initTableFromCsv[A <: AbstractTable[_], H <: HList](table: TableQuery[A], network: String, separator: Char = ',')(
+      implicit hd: HeaderDecoder[A#TableElementType],
+      g: Generic.Aux[A#TableElementType, H],
+      m: Mapper.Aux[tech.cryptonomic.conseil.util.ConfigUtil.Csv.Trimmer.type, H, H]
+  ): Future[Option[ProposalSupporters]] = {
+    val rows = ConfigUtil.Csv.readTableRowsFromCsv(table, network, separator)
+    db.run(insertWhenEmpty(table, rows)) andThen {
+      case Success(_) => logger.info(s"Written ${rows.size} ${table.baseTableRow.tableName} rows")
+      case Failure(e) => logger.error(s"Could not fill ${table.baseTableRow.tableName} table", e)
     }
+  }
```

Comment:
I'd be surprised if this method signature would require to be that complex anymore.

It should do nothing more than what is on the code before the change, that is, read the tokens and call the db.
All those implicits shouldn't be required anymore, and neither should the extra shapeless and kantan imports, apart from the one already there.

Reply:
I also think that there is some incorrect type inference that proposed the result type as
`Future[Option[ProposalSupporters]]`, trying to be too smart 😁

Reply:
Unfortunately it requires those implicits. They are generated for the specific types, so they need to be passed with the method. If I had implementation for the specific type like just for `Tables.KnownAddresses` I would not require to put it there.

Reply:
and regarding the `ProposalSupporters`, yes, that was IntelliJ helping 😄

Reply:
ooh, my bad, I didn't realize it's generic of his own... what about moving this into TezosDatabaseOperations, to keep the Lorre code more clean?

Reply:
I moved it out of there as TezosDatabaseOperations did not use `db.run()` and I did not want to introduce it there.